### PR TITLE
server: make it possible to enable ws/http only

### PR DESCRIPTION
**Pull request by niklasad1**

### CODE REVIEW

File: server/src/server.rs [hunk 1]
```diff
@@ -425,6 +433,27 @@ impl<B, L> Builder<B, L> {
...
+	/// Default: both http and ws are enabled.
+	pub fn http_only(mut self) -> Self {
+		self.settings.enable_http = true;
+		self.settings.enable_ws = false;
+		self
+	}
+
+	/// Configure the server should only serve JSON-RPC WebSocket requests.
+	///
+	/// That implies that server just denies pure HTTP request which isn't a WebSocket upgrade request
```
Comment (niklasad1) [inline server/src/server.rs:447]:
```suggestion
	/// That implies that server just denies pure HTTP requests which isn't a WebSocket upgrade request
```

File: server/src/server.rs [hunk 2]
```diff
@@ -425,6 +433,27 @@ impl<B, L> Builder<B, L> {
...
+		self
+	}
+
+	/// Configure the server should only serve JSON-RPC WebSocket requests.
+	///
+	/// That implies that server just denies HTTP requests which isn't a WebSocket upgrade request
+	/// except if you have enabled middleware which are processed before this check occurs.
+	///
+	/// Default: both http and ws are enabled.
+	pub fn ws_only(mut self) -> Self {
```
Comment (niklasad1) [inline server/src/server.rs:451]: @lexnv this is a bit opinionated but I would like to have it for consistency.

File: server/src/server.rs [hunk 3]
```diff
@@ -425,6 +433,27 @@ impl<B, L> Builder<B, L> {
...
+	pub fn http_only(mut self) -> Self {
+		self.settings.enable_http = true;
+		self.settings.enable_ws = false;
+		self
+	}
+
+	/// Configure the server should only serve JSON-RPC WebSocket requests.
+	///
+	/// That implies that server just denies HTTP requests which isn't a WebSocket upgrade request
+	/// except if you have enabled middleware which are processed before this check occurs.
```
Comment (niklasad1) [inline server/src/server.rs:448]:
```suggestion
```

File: server/src/server.rs (hunk 2 above)
Comment (niklasad1) [reply server/src/server.rs:450]: this would disable all JSON-RPC HTTP requests just reply 403 on all HTTP requests (including middleware as well)

**Review by lexnv — APPROVED** [review]
LGTM! 👍

File: server/src/server.rs [hunk 4]
```diff
@@ -425,6 +433,26 @@ impl<B, L> Builder<B, L> {
...
+	/// Configure the server to only serve JSON-RPC HTTP requests.
+	///
+	/// Default: both http and ws are enabled.
+	pub fn http_only(mut self) -> Self {
+		self.settings.enable_http = true;
+		self.settings.enable_ws = false;
+		self
+	}
+
+	/// Configure the server should only serve JSON-RPC WebSocket requests.
```
Comment (lexnv) [inline server/src/server.rs:445]:
```suggestion
	/// Configure the server to only serve JSON-RPC WebSocket requests.
```

**Review by jsdw — APPROVED** [review]
Nice!

**Pull request merged**